examples/SharedMemory/PhysicsServerCommandProcessor.cpp

```diff
@@ -4832,9 +4895,17 @@ bool PhysicsServerCommandProcessor::processRequestRaycastIntersectionsCommand(co
 
 	const int numCommandRays = clientCmd.m_requestRaycastIntersections.m_numCommandRays;
 	const int numStreamingRays = clientCmd.m_requestRaycastIntersections.m_numStreamingRays;
-	const int numThreads = clientCmd.m_requestRaycastIntersections.m_numThreads;
+	const int totalRays = numCommandRays+numStreamingRays;
+	int numThreads = clientCmd.m_requestRaycastIntersections.m_numThreads;
+	if (numThreads == 0) {
+		// When 0 is specified, Bullet can decide how many threads to use.
+		// About 16 rays per thread seems to work reasonably well.
+		numThreads = btMax(1, totalRays / 16);
```

Comment:
Is there any limit on the number of threads? We don't want too many threads, I think a limit of say 4 to 32 threads would be good.

Reply:
This is already limited to min(BT_NUM_THREADS, number of virtual cores) where BT_NUM_THREADS is 64.
Happy to add an additional limit if you still want it